### PR TITLE
runelite-api: Add queries

### PR DESCRIPTION
**Pull request by devinfrench**

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/Query.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+package net.runelite.api;
+
+import java.util.function.Predicate;
+
+public abstract class Query<T>
+{
+	protected Predicate<T> predicate;
+
+	protected Query()
```
Comment (Adam-) [inline runelite-api/src/main/java/net/runelite/api/Query.java]: protected constructor is unnecessary, it is an abstract class

**Comment by Adam-** [comment]
Both player and npc extend actor which has a getName, so you can make an abstract ActorQuery to contain the name filter logic.

**Comment by devinfrench** [comment]
Having a protected constructor prevents creating a Query object outside of the package. Being able to create a Query object that is not specific to a type is unwanted behavior.

**Comment by devinfrench** [comment]
Open to different naming convention for filters if anyone has suggestions.

**Pull request merged**